pkg/handler/collector/oci/oci.go

```diff
@@ -183,89 +208,192 @@ func (o *ociCollector) getTagsAndFetch(ctx context.Context, repo string, tags []
 // A workaround for this would be to run the collector again with a specific tag without polling and ingest like normal
 func (o *ociCollector) fetchOCIArtifacts(ctx context.Context, repo string, rc *regclient.RegClient, image ref.Ref, docChannel chan<- *processor.Document) error {
 	logger := logging.FromContext(ctx)
-
 	// attempt to request only the headers, avoids Docker Hub rate limits
 	m, err := rc.ManifestHead(ctx, image)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed retrieving manifest head: %w", err)
 	}
 
 	if m.IsList() {
 		m, err := rc.ManifestGet(ctx, image)
 		if err != nil {
-			return err
+			return fmt.Errorf("failed retrieving manifest: %w", err)
 		}
+
 		pl, _ := manifest.GetPlatformList(m)
+		logger.Infof("%s is manifest list with %d platforms", image.Reference, len(pl))
+
+		// Use goroutines to fetch platforms concurrently
+		// Create a channel to collect errors from goroutines
+		errorChan := make(chan error, len(pl))
+		var wg sync.WaitGroup
+
 		for _, p := range pl {
-			desc, err := manifest.GetPlatformDesc(m, p)
+			// Increment the WaitGroup counter
+			wg.Add(1)
+			go func(p *platform.Platform) {
+				defer wg.Done() // Decrement the WaitGroup counter when done
+				logger.Infof("Fetching platform %s", p)
+				desc, err := manifest.GetPlatformDesc(m, p)
+				if err != nil {
+					errorChan <- fmt.Errorf("failed retrieving platform specific digest: %w", err)
+				}
+				platformImage := ref.Ref{
+					Scheme:     image.Scheme,
+					Registry:   image.Registry,
+					Repository: image.Repository,
+					Digest:     desc.Digest.String(),
+				}
+				logger.Infof("Fetching %s for platform %s", platformImage.Digest, desc.Platform)
+				if err := o.fetchOCIArtifacts(ctx, repo, rc, platformImage, docChannel); err != nil {
+					errorChan <- fmt.Errorf("failed retrieving platform specific digest: %w", err)
```

Comment:
Ensure that error messages are descriptive and consistent.

 For example, fmt.Errorf("failed retrieving platform specific digest: %w", err) is used in two different places, which might make troubleshooting difficult.
 
  Differentiate error messages to pinpoint issues effectively.